money-monkey-web/app/lib/validation.ts

```
@@ -15,7 +15,14 @@ export const NewExpenseSchema: Yup.ObjectSchema<NewExpense> = Yup.object({
       const numericValue = value ? Number.parseFloat(value.slice(1)) : 0;
       return !Number.isNaN(numericValue) && numericValue >= 0.01;
     }),
-  date: Yup.date().required("Required!"),
+  date: Yup.string()
+    .required("Required!")
+    .matches(/^\d{4}-\d{2}-\d{2}$/, "Invalid date format!")
+    .test("is-valid-date", "Invalid date!", (value) => {
+      if (!value) return false;
+      const date = new Date(value);
+      return date instanceof Date && !isNaN(date.getTime());
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)